src/ui/pages/SendTransaction/SendTransaction.tsx

```diff
@@ -83,6 +94,186 @@ import { TransactionAdvancedView } from './TransactionAdvancedView';
+) {
+  interface PaymasterParamsResponse {
+    data: {
+      eligible: boolean;
+      paymasterParams: {
+        paymaster: string;
+        paymasterInput: string;
+      };
+      chargesData: {
+        amount: number;
+        deadline: string;
+        eta: null;
+      };
+    };
+    errors?: null | { title: string; detail: string }[];
+  }
+  type Request = {
+    from: string;
+    to: string;
+    nonce: string;
+    chainId: string;
+    gas: string;
+    gasPerPubdataByte: string;
+    maxFee: string;
+    maxPriorityFee: string;
+    value: string;
+    data: string;
+  };
+  const transaction = assertTransactionProps(incomingTransaction);
+  const params: Request = {
+    from: transaction.from,
+    to: transaction.to,
+    nonce: valueToHex(transaction.nonce),
+    chainId: normalizeChainId(transaction.chainId),
+    gas: valueToHex(transaction.gasLimit),
+    gasPerPubdataByte,
+    data: valueToHex(transaction.data),
+    maxFee: valueToHex(transaction.maxFeePerGas),
+    maxPriorityFee: valueToHex(transaction.maxPriorityFeePerGas),
+    value: valueToHex(transaction.value),
+  };
+  const { data } = await ZerionAPI.get<PaymasterParamsResponse>(
+    `/paymaster/get-params/v1?${new URLSearchParams({
+      ...params,
+      backend_env: 'zero',
+    })}`
+  );
+  return data;
+}
+
+async function configureTransactionToSign<T extends IncomingTransaction>(
+  transaction: T,
+  {
+    transactionAction,
+    allowanceQuantityBase,
+    chain,
+    networks,
+    from,
+    configuration,
+    chainGasPrices,
+  }: {
+    transactionAction: TransactionAction;
+    allowanceQuantityBase: string | null;
+    chain: Chain;
+    networks: Networks;
+    from: string;
+    configuration: CustomConfiguration;
+    chainGasPrices: ChainGasPrice | null;
+  }
+): Promise<PartiallyRequired<T, 'chainId' | 'from'>> {
+  let tx = transaction as T | IncomingTransaction;
+
+  const chainId = transaction.chainId || networks.getChainId(chain);
+  invariant(chainId, 'Could not resolve chainId for tx to sign');
+  tx.chainId = chainId;
+
+  if (transactionAction.type === 'approve' && allowanceQuantityBase) {
+    tx = await walletPort.request('createApprovalTransaction', {
+      chain: chain.toString(),
+      contractAddress: transactionAction.contractAddress,
+      allowanceQuantityBase,
+      spender: transactionAction.spenderAddress,
+    });
+    tx.chainId = networks.getChainId(chain) ?? undefined;
+    tx.from = from;
+    const gas = await estimateGas(tx, networks);
+    tx.gasLimit = gas;
+  }
+
+  if (tx.value == null) {
+    tx.value = '0x0';
+  }
+
+  if (tx.nonce == null) {
+    const { value: nonce } = await uiGetBestKnownTransactionCount({
+      address: tx.from || from,
+      chain,
+      networks,
+      defaultBlock: 'pending',
+    });
+    tx = { ...tx, nonce };
+  }
+
+  tx = applyConfiguration(tx, configuration, chainGasPrices);
+
+  return tx as PartiallyRequired<T, 'chainId' | 'from'>;
```

Comment:
'chainId' | 'from' | 'nonce' ?

Reply:
Overall, as I see, we help only approve transactions to set `from` value

Reply:
Hm, I guess so. We don't rely (in code) though on it being there